pages/api/socketio.js

```diff
@@ -29,7 +33,7 @@ const ioHandler = (req, res) => {
             socketMessage = extractImageFromInputOutputMessage(msg);
           }
 
-          socket.broadcast.emit(emitName, socketMessage)
+          socket.to(msg["source"]).emit(logType, socketMessage)
```

Comment:
Are you sure that this is not happening when the emit name is only log type? ->  Let's say we have two routines - A, B. Both of them log input and output. A user is connecting with two tabs, which means two sockets are opening and two routines are sending input/output. The emit message is the same so both of them get both input/output of both routine.

Reply:
I think that specifying room name `to(msg["source)` handles your described use-case.
It should be tested.